core/vulkan/vk_virtual_swapchain/cc/swapchain.cpp

```diff
@@ -355,13 +408,33 @@ vkQueuePresentKHR(VkQueue queue, const VkPresentInfoKHR* pPresentInfo) {
         nullptr                                            // pSemaphores
     };
 
-    res |= GetGlobalContext().GetQueueData(queue)->vkQueueSubmit(
-        queue, 1, &submitInfo, swp->GetFence(image_index));
-    res |= swp->PresentToSurface(queue, image_index);
+    res = EXPECT_SUCCESS(GetGlobalContext().GetQueueData(queue)->vkQueueSubmit(
+        queue, 1, &submitInfo, swp->GetFence(image_index)));
+
+    if (res != VK_SUCCESS) {
+      break;
+    }
+
+    res = swp->PresentToSurface(queue, image_index);
```

Comment:
Should EXPECT_SUCCESS be used here too?

Reply:
I decided to focus just on Vulkan calls, and just on those that I think should always succeed. Possibly this could fail and it would not be unexpected. Not sure. The EXPECT_SUCCESS macro will not do the break though, so we need that.

Reply:
Makes sense!